pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -271,12 +271,20 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Fn:           flow.TaskFn(botanist.DeployEtcd).RetryUntilTimeout(defaultInterval, helper.GetEtcdDeployTimeout(o.Shoot, defaultTimeout)),
 			Dependencies: flow.NewTaskIDs(initializeSecretsManagement, deployCloudProviderSecret, waitUntilBackupEntryInGardenReconciled, waitUntilEtcdBackupsCopied),
 		})
-		_ = g.Add(flow.Task{
+		destroySourceBackupEntry = g.Add(flow.Task{
 			Name:         "Destroying source backup entry",
 			Fn:           botanist.DestroySourceBackupEntry,
 			SkipIf:       !allowBackup,
 			Dependencies: flow.NewTaskIDs(deployETCD),
 		})
+		_ = g.Add(flow.Task{
```

Comment:
I was wondering whether we have to wait for https://github.com/gardener/gardener-extension-provider-azure/issues/793 to be fixed first before adding such a wait in the flow. 
IIUC, such wait would cause the restore operation to fail because the BackupEntry won't be deleted due to the above issue.
Alternatively, we could comment the code with appropriate TODO and uncomment it when the provider-azure issue is fixed.
@plkokanov WDYT?

Reply:
Hmm now that I gave it a second thought, you're probably right. If we merge it now, the `restore` phase will break and the shoot will not be fully migrated and therefore not deleted until the azure extension is also fixed so we might have some extra shoots laying around in our test infrastructure. 
We could alternatively hold this PR until azure is fixed instead of adding a `TODO` and just open PRs to fix the separate provider extensions as discussed in https://github.com/gardener/gardener/issues/8730